tests/utils/test_file_helpers.py

```diff
@@ -183,3 +184,86 @@ def test_format_name_with_dist_and_time():
 
 def test_ensure_folder_is_empty(tmpdir: pathlib.Path):
     ensure_folder_is_empty(tmpdir)
+
+
+def test_ensure_folder_has_no_conflicting_files(tmpdir: pathlib.Path):
+    timestamp = Timestamp(2, 7, 1, 15, 3, 31, 7)
+    run_name = "blazing-unicorn"
+    filename = f"{run_name}-ep{{epoch}}-batch{{batch}}-tie{{token_in_epoch}}-rank{{rank}}.pt"
+
+    # Ignore empty folder
+    ensure_folder_has_no_conflicting_files(tmpdir, filename, timestamp)
+
+    # Ignore timestamps in past
+    with open(os.path.join(tmpdir, f'{run_name}-ep1-batch3-tie6-rank0.pt'), 'w') as f:
```

Comment:
nit: can each of these tests be seperate? Feel free to use pytest parameterization or a shared helper function...